### PR TITLE
[firebase_auth] Migrate to Android v2 embedder.

### PR DESCRIPTION
**Pull request by zhenqiu1101**
## Description

Migrate firebase_auth plugin to android v2 embedder. Keep it backward compatible with v1.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by zhenqiu1101** [comment]
@mehmetf @cyanglaz

File: packages/firebase_auth/firebase_auth/example/pubspec.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@ dev_dependencies:
   flutter_driver:
     sdk: flutter
   test: any
+  e2e: ^0.2.1
```
Comment (mehmetf) [inline packages/firebase_auth/firebase_auth/example/pubspec.yaml:20]: I don't see a new test file, why is there a new dependency?

File: packages/firebase_auth/firebase_auth/test/firebase_auth_e2e.dart [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2020, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:e2e/e2e.dart';
+
+void main() {
+  E2EWidgetsFlutterBinding.ensureInitialized();
+}
```
Comment (cyanglaz) [inline packages/firebase_auth/firebase_auth/test/firebase_auth_e2e.dart:9]: Are there any tests that can be added here?

File: packages/firebase_auth/firebase_auth/test_driver/firebase_auth_e2e_test.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2020, the Chromium project authors.  Please see the AUTHORS file
```
Comment (cyanglaz) [inline packages/firebase_auth/firebase_auth/test_driver/firebase_auth_e2e_test.dart:1]: I think this file needs to be moved to the example app. Does it work if you do `flutter drive test/firebase_auth_e2e.dart` now?

Comment (zhenqiu1101) [reply packages/firebase_auth/firebase_auth/test_driver/firebase_auth_e2e_test.dart:1]: I can't due to the gradle issue we've talked about. But if I moved it to example, it complains no test file found.

File: packages/firebase_auth/firebase_auth/test/firebase_auth_e2e.dart (hunk 2 above)
Comment (zhenqiu1101) [reply packages/firebase_auth/firebase_auth/test/firebase_auth_e2e.dart:9]: I doubt if we can test any real sign in behavior? It's like the Google sign in plugin.

**Comment by cyanglaz** [comment]
@collinjackson There seems to be flutter driver tests set up in this plugin to test the functionalities. Do we still need to convert them to e2e?
Also, how do those integration tests run in CI?

**Pull request merged**